data/transactions/verify/txn.go

```diff
@@ -99,59 +99,43 @@ func (g *GroupContext) Equal(other *GroupContext) bool {
 		g.minAvmVersion == other.minAvmVersion
 }
 
-// Txn verifies a SignedTxn as being signed and having no obviously inconsistent data.
+// txnBatchPrep verifies a SignedTxn having no obviously inconsistent data.
 // Block-assembly time checks of LogicSig and accounting rules may still block the txn.
-func Txn(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContext) error {
-	batchVerifier := crypto.MakeBatchVerifier()
-
-	if err := TxnBatchVerify(s, txnIdx, groupCtx, batchVerifier); err != nil {
-		return err
-	}
-
-	// this case is used for comapact certificate where no signature is supplied
-	if batchVerifier.GetNumberOfEnqueuedSignatures() == 0 {
-		return nil
-	}
-	return batchVerifier.Verify()
-}
-
-// TxnBatchVerify verifies a SignedTxn having no obviously inconsistent data.
-// Block-assembly time checks of LogicSig and accounting rules may still block the txn.
-// it is the caller responsibility to call batchVerifier.verify()
-func TxnBatchVerify(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContext, verifier *crypto.BatchVerifier) error {
+// it is the caller responsibility to call batchVerifier.Verify()
+func txnBatchPrep(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContext, verifier *crypto.BatchVerifier) error {
 	if !groupCtx.consensusParams.SupportRekeying && (s.AuthAddr != basics.Address{}) {
-		return errors.New("nonempty AuthAddr but rekeying not supported")
+		return errors.New("nonempty AuthAddr but rekeying is not supported")
 	}
 
 	if err := s.Txn.WellFormed(groupCtx.specAddrs, groupCtx.consensusParams); err != nil {
 		return err
 	}
 
-	return stxnVerifyCore(s, txnIdx, groupCtx, verifier)
+	return stxnCoreChecks(s, txnIdx, groupCtx, verifier)
 }
 
 // TxnGroup verifies a []SignedTxn as being signed and having no obviously inconsistent data.
 func TxnGroup(stxs []transactions.SignedTxn, contextHdr bookkeeping.BlockHeader, cache VerifiedTransactionCache, ledger logic.LedgerForSignature) (groupCtx *GroupContext, err error) {
 	batchVerifier := crypto.MakeBatchVerifier()
 
-	if groupCtx, err = TxnGroupBatchVerify(stxs, contextHdr, cache, ledger, batchVerifier); err != nil {
+	if groupCtx, err = txnGroupBatchPrep(stxs, contextHdr, ledger, batchVerifier); err != nil {
 		return nil, err
 	}
 
```

Comment:
since we removed the check for `GetNumberOfEnqueuedSignatures` here I think we should be consistent with `PaysetGroups` function
https://github.com/algorand/go-algorand/blob/5be155b2cf75bf461d52333f5b4b26e545a2bf4c/data/transactions/verify/txn.go#L413

Reply:
maybe here as well?
https://github.com/algorand/go-algorand/blob/5be155b2cf75bf461d52333f5b4b26e545a2bf4c/data/transactions/verify/txn.go#L254